chain/src/main/scala/org/bitcoins/chain/blockchain/BaseBlockChain.scala

```diff
@@ -55,7 +55,8 @@ private[blockchain] trait BaseBlockChain {
 
   /** Splits the blockchain at the header, returning a new blockchain where the best tip is the given header */
   def fromHeader(header: BlockHeaderDb): Option[Blockchain] = {
-    val headerIdxOpt = headers.zipWithIndex.find(_._1 == header)
+    // we don't want to create another copy of headers vector in zipWithIndex, so we wrap it in a Stream
+    val headerIdxOpt = headers.toStream.zipWithIndex.find(_._1 == header)
```

Comment:
`.toStream` seems to be deprecated

```scala
  @deprecated("Use .to(LazyList) instead of .toStream", "2.13.0")
  @`inline` final def toStream: immutable.Stream[A] = to(immutable.Stream)
```

Reply:
Should we deprecate support for Scala 2.11 and 2.12 then? There's no LazyList in their standard libraries.

Reply:
Take a look at the top of this file. We have separate file structures for `2.11` `2.12` and `2.13` because of this. We should probably move things that use `.toStream()` into the corresponding scala version